openedx_events/data.py

```diff
@@ -37,7 +48,12 @@ class EventsMetadata:
     minorversion = attr.ib(type=int, converter=attr.converters.default_if_none(0))
     source = attr.ib(type=str, init=False)
     sourcehost = attr.ib(type=str, init=False)
-    time = attr.ib(type=datetime, init=False)
+    current_utc_time = datetime.now(timezone.utc)
+    # Note: cannot use default_if_none for time, because it caches the time the class is loaded.
+    time = attr.ib(
```

Comment:
Can we do something like 
```
time = attr.ib(
    type=datetime,
    default = attr.Factory(lambda: datetime.now(timezone.utc), takes_self=False),
    validator=attr.validators.optional([attr.validators.instance_of(datetime), _ensure_utc_time]),
)
```
?
Asking because I think future us might be confused and/or annoyed at constantly having to put time=None on things. FWIW I tried something like this locally and because it uses a callable it didn't cache the time.

Reply:
(and honestly if I had my way we'd do the same thing for minorversion)

Reply:
I found the right combo. I didn't know about `attr.Factory`, and had just tried passing `default_if_none` a lamba orginally, and that didn't work. Passing it the factory works.

Reply:
fancy nesting! thanks